torch/csrc/inductor/aoti_eager/kernel_meta_info.h

```diff
@@ -0,0 +1,42 @@
+#if !defined(C10_MOBILE) && !defined(ANDROID)
+#pragma once
+
+#include <ATen/ATen.h>
+#include <c10/core/SymIntArrayRef.h>
+
+#include <string>
+
+namespace torch::inductor {
+
+class TensorChecker;
+
+struct TensorMetaInfo {
```

Comment:
Add docs about caching strategy and how this related to guards.

Reply:
Added docs.